src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -1249,6 +1283,32 @@ static ArrayBuilder<TypeParameterDiagnosticInfo> ensureUseSiteDiagnosticsBuilder
             return false;
         }
 
+        private static bool SelfOrBaseHasStaticAbstractMember(TypeWithAnnotations constraintType, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            Debug.Assert(constraintType.Type.IsInterfaceType());
+
+            Func<Symbol, bool> predicate = static m => m.IsStatic && m.IsAbstract;
+            var definition = (NamedTypeSymbol)constraintType.Type.OriginalDefinition;
+
+            if (definition.GetMembersUnordered().Any(predicate))
```

Comment:
Consider using:

https://github.com/dotnet/roslyn/blob/b90760f73f03449a6882e3e6ed0aa9aaff379333/src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs#L378

(Not sure if it matters in this specific case)

Reply:
I don't believe that matters here, as the predicate is a static lambda and needs no inputs. That method is for when the predicate needs extra inputs from the current context, so that we can still make the predicate a static lambda in those cases.

---
In reply to: [606304684](https://github.com/dotnet/roslyn/pull/52366#discussion_r606304684) [](ancestors = 606304684)

Reply:
> I don't believe that matters here, as the predicate is a static lambda and needs no inputs. That method is for when the predicate needs extra inputs from the current context, so that we can still make the predicate a static lambda in those cases.

Agree.

---
In reply to: [606327930](https://github.com/dotnet/roslyn/pull/52366#discussion_r606327930) [](ancestors = 606327930,606304684)